launcher/tasks/ConcurrentTask.cpp

```diff
@@ -108,27 +109,33 @@ void ConcurrentTask::clear()
     m_failed.clear();
     m_queue.clear();
 
-    m_aborted = false;
-
     m_progress = 0;
     m_stepProgress = 0;
 }
 
-void ConcurrentTask::startNext()
+void ConcurrentTask::executeNextSubTask()
 {
-    if (m_aborted || m_doing.count() > m_total_max_size)
+    if (!isRunning()) {
         return;
-
-    if (m_queue.isEmpty() && m_doing.isEmpty() && !wasSuccessful()) {
-        emitSucceeded();
+    }
+    if (m_queue.isEmpty()) {
+        if (m_doing.isEmpty()) {
+            // if (m_failed.isEmpty())
+            emitSucceeded();
+            // else
+            // emitFailed(tr("One or more subtasks failed"));
```

Comment:
I'm considering merging this PR with https://github.com/PrismLauncher/PrismLauncher/pull/1364 as both modify the ConncurentTask in a way that creates a need for heavy testing

Reply:
decided to merge the two PRs.